lib/Graph/Grad.cpp

```diff
@@ -166,10 +168,10 @@ Function *glow::differentiate(Function *F, const TrainingConfig &conf,
           F->getParent()->uniqueTypeWithNewShape(TNInputType, BRAInputDims);
 
       auto *RN =
-          new ReshapeNode(TN->getName().str() + ".grad.reshape", BRAInputType,
+          new ReshapeNode(TN->getName().str() + "_grad_reshape", BRAInputType,
```

Comment:
It seems like it would be also beneficial to have another helper(s) to create a compound name from name segments delimited by `_`, e.g. `getName(x, "grad_reshape")` or even  `getName(x, "grad", "reshape")`. It avoids an explicit string manipulation and hides the implementation details like the use of `_` as a delimiter.